core/src/main/java/org/apache/iceberg/avro/AvroSchemaUtil.java

```diff
@@ -404,4 +405,21 @@ private static String sanitize(char character) {
     }
     return "_x" + Integer.toHexString(character).toUpperCase();
   }
+
+  public static Schema removeFields(Schema.Field field) {
```

Comment:
The implementation here seems very specific to a case and not useful as a generic method. Maybe we should move it into the `PruneColumns` class.

Reply:
Yep sure, it's not very common